File: govern/data-security/ranger-2.3.0/security-admin/src/main/java/org/apache/ranger/view/VXAsset.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

 package org.apache.ranger.view;

/**
 * Asset
 *
 */

import javax.xml.bind.annotation.XmlRootElement;

import org.apache.ranger.common.AppConstants;
import org.codehaus.jackson.annotate.JsonAutoDetect;
import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility;
import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonAutoDetect(getterVisibility=Visibility.NONE, setterVisibility=Visibility.NONE, fieldVisibility=Visibility.ANY)
@JsonSerialize(include=JsonSerialize.Inclusion.NON_NULL )
@JsonIgnoreProperties(ignoreUnknown=true)
@XmlRootElement
public class VXAsset extends VXDataObject implements java.io.Serializable {
	private static final long serialVersionUID = 1L;


	/**
	 * Name
	 */
	protected String name;
	/**
	 * Description
	 */
	protected String description;
	/**
	 * Status
	 * This attribute is of type enum CommonEnums::ActiveStatus
	 */
	protected int activeStatus;
	/**
	 * Type of asset
	 * This attribute is of type enum CommonEnums::AssetType
	 */
	protected int assetType = AppConstants.ASSET_UNKNOWN;
	/**
	 * Config in json format
	 */
	protected String config;
	/**
	 * Support native authorization
	 */
	protected boolean supportNative = false;

	/**
	 * Default constructor. This will set all the attributes to default value.
	 */
	public VXAsset ( ) {
		activeStatus = 0;
		assetType = AppConstants.ASSET_UNKNOWN;
	}

	/**
	 * This method sets the value to the member attribute <b>name</b>.
	 * You cannot set null to the attribute.
	 * @param name Value to set member attribute <b>name</b>
	 */
	public void setName( String name ) {
		this.name = name;
	}

	/**
	 * Returns the value for the member attribute <b>name</b>
	 * @return String - value of member attribute <b>name</b>.
	 */
	public String getName( ) {
		return this.name;
	}

	/**
	 * This method sets the value to the member attribute <b>description</b>.
	 * You cannot set null to the attribute.
	 * @param description Value to set member attribute <b>description</b>
	 */
	public void setDescription( String description ) {
		this.description = description;
	}

	/**
	 * Returns the value for the member attribute <b>description</b>
	 * @return String - value of member attribute <b>description</b>.
	 */
	public String getDescription( ) {
		return this.description;
	}

	/**
	 * This method sets the value to the member attribute <b>activeStatus</b>.
	 * You cannot set null to the attribute.
	 * @param activeStatus Value to set member attribute <b>activeStatus</b>
	 */
	public void setActiveStatus( int activeStatus ) {
		this.activeStatus = activeStatus;
	}

	/**
	 * Returns the value for the member attribute <b>activeStatus</b>
	 * @return int - value of member attribute <b>activeStatus</b>.
	 */
	public int getActiveStatus( ) {
		return this.activeStatus;
	}

	/**
	 * This method sets the value to the member attribute <b>assetType</b>.
	 * You cannot set null to the attribute.
	 * @param assetType Value to set member attribute <b>assetType</b>
	 */
	public void setAssetType( int assetType ) {
		this.assetType = assetType;
	}

	/**
	 * Returns the value for the member attribute <b>assetType</b>
	 * @return int - value of member attribute <b>assetType</b>.
	 */
	public int getAssetType( ) {
		return this.assetType;
	}

	/**
	 * This method sets the value to the member attribute <b>config</b>.
	 * You cannot set null to the attribute.
	 * @param config Value to set member attribute <b>config</b>
	 */
	public void setConfig( String config ) {
		this.config = config;
	}

	/**
	 * Returns the value for the member attribute <b>config</b>
	 * @return String - value of member attribute <b>config</b>.
	 */
	public String getConfig( ) {
		return this.config;
	}

	/**
	 * This method sets the value to the member attribute <b>supportNative</b>.
	 * You cannot set null to the attribute.
	 * @param supportNative Value to set member attribute <b>supportNative</b>
	 */
	public void setSupportNative( boolean supportNative ) {
		this.supportNative = supportNative;
	}

	/**
	 * Returns the value for the member attribute <b>supportNative</b>
	 * @return boolean - value of member attribute <b>supportNative</b>.
	 */
	public boolean isSupportNative( ) {
		return this.supportNative;
	}

	@Override
	public int getMyClassType( ) {
	    return AppConstants.CLASS_TYPE_XA_ASSET;
	}

	/**
	 * This return the bean content in string format
	 * @return formatedStr
	*/
	public String toString( ) {
		String str = "VXAsset={";
		str += super.toString();
		str += "name={" + name + "} ";
		str += "description={" + description + "} ";
		str += "activeStatus={" + activeStatus + "} ";
		str += "assetType={" + assetType + "} ";
		str += "config={" + config + "} ";
		str += "supportNative={" + supportNative + "} ";
		str += "}";
		return str;
	}
}

File: govern/data-security/ranger-2.3.0/security-admin/src/main/webapp/scripts/models/VXAsset.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

 
define(function(require){
	'use strict';	

	var VXAssetBase	= require('model_bases/VXAssetBase');
	var XAUtils		= require('utils/XAUtils');
	var XAEnums		= require('utils/XAEnums');
	var localization= require('utils/XALangSupport');

	var VXAsset = VXAssetBase.extend(
	/** @lends VXAsset.prototype */
	{
		/**
		 * VXAsset initialize method
		 * @augments VXAssetBase
		 * @constructs
		 */
		initialize: function() {
			this.modelName = 'VXAsset';
			this.bindErrorEvents();
		},
		/**
		 * @function schema
		 * This method is meant to be used by UI,
		 * by default we will remove the unrequired attributes from serverSchema
		 */

		schemaBase : function(){
			var attrs = _.omit(this.serverSchema, 'id', 'createDate', 'updateDate', "version",
					"displayOption", "permList", "forUserId", "status", "priGrpId",
					"updatedBy","isSystem");

			_.each(attrs, function(o){
				o.type = 'Hidden';
			});

			// Overwrite your schema definition here
			return _.extend(attrs,{
				name : {
					type		: 'Text',
					title		: 'Repository Name *',
					validators	: ['required'],
					editorAttrs 	:{ maxlength: 255},
				},
				description : {
					type		: 'TextArea',
					title		: 'Description',
					validators	: []
				},
				activeStatus : {
			        type: 'Radio',
					options : function(callback, editor){
						var activeStatus = _.filter(XAEnums.ActiveStatus,function(m){return m.label != 'Deleted'});
						var nvPairs = XAUtils.enumToSelectPairs(activeStatus);
						callback(_.sortBy(nvPairs, function(n){ return !n.val; }));
					}
				},
				assetType : {
					type : 'Select',
					options : function(callback, editor){
						var assetTypes = _.filter(XAEnums.AssetType,function(m){return m.label != 'Unknown'});
						var nvPairs = XAUtils.enumToSelectPairs(assetTypes);
						callback(nvPairs);
					},
					title : localization.tt('lbl.assetType'),
					editorAttrs:{'disabled' : true}
				}

			});
		},

		/** This models toString() */
		toString : function(){
			return this.get('name');
		},
		propertiesNameMap : {
			userName : "username",
			passwordKeytabfile : "password",
			fsDefaultName : "fs.default.name",
			authorization : "hadoop.security.authorization",
			authentication : "hadoop.security.authentication",
			auth_to_local : "hadoop.security.auth_to_local",
			datanode : "dfs.datanode.kerberos.principal",
			namenode : "dfs.namenode.kerberos.principal",
			secNamenode : "dfs.secondary.namenode.kerberos.principal",
			hadoopRpcProtection : "hadoop.rpc.protection",
			//hive
			driverClassName : "jdbc.driverClassName",
			url	: "jdbc.url",
			
			masterKerberos     		: 'hbase.master.kerberos.principal',
			rpcEngine 				: 'hbase.rpc.engine',
			rpcProtection	 		: 'hbase.rpc.protection',
			securityAuthentication  : 'hbase.security.authentication',
			zookeeperProperty 		: 'hbase.zookeeper.property.clientPort',
			zookeeperQuorum 		: 'hbase.zookeeper.quorum',
			//hbase
			zookeeperZnodeParent	: 'zookeeper.znode.parent',
			//knox
			knoxUrl					:'knox.url',
			//Storm
			nimbusUrl				:'nimbus.url',
			
			commonnameforcertificate: 'commonNameForCertificate'
		}

	}, {
		// static class members
	});

    return VXAsset;
	
});


